Documentation updates (#122)
spine-generic/spine-generic: docs/documentation.rst

    else
      fname_seg=${fname_data%%".nii.gz"*}_seg.nii.gz${fname_data##*".nii.gz"}
      fname_seg_dest=${PATH_SEGMANUAL}/${file%%".nii.gz"*}_seg-manual.nii.gz${file##*".nii.gz"}
    fi
    # Copy file to PATH_SEGMANUAL
    cp $fname_seg $fname_seg_dest
    # Launch FSLeyes
    echo "In FSLeyes, click on 'Edit mode', correct the segmentation, then save it with the same name (overwrite)."
    fsleyes -yh $fname_data $fname_seg_dest -cm red
  done

- Make this script executable:

  .. code-block:: bash

     chmod 775 fix_seg.sh

- In the QC report, enter the string "deepseg" to only display segmentation results.
- Review all segmentations. Use the keyboard shortcuts up/down arrow to switch between
  subjects and the left arrow to toggle overlay.
- If you spot *major* issues with the segmentation (e.g. noticeable leaking or under-segmentation that extends over several slices),
  add the image name in the variable array ``FILES`` in the script.
- If the data quality is too low to be interpreted (too blurry, large artifacts),
  add the image file name to the variable ``TO_EXCLUDE`` in the file ``parameters.sh``,

>   which will be used in the next processing iteration.

Author: This point was out-of-date because `generate_figure.py` no longer requires `parameters.sh` file. Subjects to remove and sites to exclude are defined directly inside `generate_figure.py` script. Is it okay for future? Because now, user has to modify the script. Or do we plan to create yaml file containing this problematic sites/subjects/segmentations and pass this yaml file as input to this script?

Reviewer: Yes, yaml would be nice indeed. Can you pls open an issue.

A subtlety: it is possible to exclude subject during processing by passing -exclude-list in sct run batch 

For now you can update the text to conform to the current behaviour

feature of the QC report to quickly jump to segmentations or labeling
results.

Segmentation and vertebral labeling
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you spot segmentation or labeling issues, manually fix them using the procedure described

Reviewer: Can you open an issue that Video tutorials have to be redone

below. Also see the video tutorial after the procedure.

- Go to the ``results/data`` folder.
- In the QC report, enter the string "deepseg" to only display segmentation results or the string "vertebrae" to only
  display vertebral labeling.
- Review all spinal cord and gray matter segmentations and vertebral labeling. Use the keyboard shortcuts up/down arrow
  to switch between subjects and the left arrow to toggle overlay.
- If you spot *major* issues with the segmentation (e.g. noticeable leaking or under-segmentation that extends over
  several slices) or wrong labeling, add the image name into the yaml file as in an example below:

.. Hint::
    You can create yaml file easily using your text editor (vim, nano, atom, ...).

::

    FILES_SEG:
    - sub-amu01_T1w_RPI_r.nii.gz
    - sub-amu01_T2w_RPI_r.nii.gz
    - sub-cardiff02_dwi_moco_dwi_mean.nii.gz
    FILES_GMSEG:
    - sub-amu01_T2star_rms.nii.gz

>     FILES_LABEL:

Reviewer: The files for vertebral labeling should be the image that the labeling is done on.

Author: Okay, but in help of `manual_correction.py` script are also listed sub ID, not files that labeling is done on:

https://github.com/spine-generic/spine-generic/blob/4a2ec8b194037107a2fc63373853efce54b67885/spinegeneric/cli/manual_correction.py#L55
https://github.com/spine-generic/spine-generic/blob/4a2ec8b194037107a2fc63373853efce54b67885/spinegeneric/cli/manual_correction.py#L56

Reviewer: > Okay, but in help of `manual_correction.py` script are also listed sub ID, not files that labeling is done on:
> 
> https://github.com/spine-generic/spine-generic/blob/4a2ec8b194037107a2fc63373853efce54b67885/spinegeneric/cli/manual_correction.py#L55
> 
> 
> https://github.com/spine-generic/spine-generic/blob/4a2ec8b194037107a2fc63373853efce54b67885/spinegeneric/cli/manual_correction.py#L56

Can you pls fix it— thanks!

Author: Fixed in https://github.com/spine-generic/spine-generic/pull/127. You can merge it into master.

   <div style="position: relative; padding-bottom: 5%; height: 0; overflow: hidden; max-width: 100%; height: auto;">
     <iframe width="700" height="394" src="https://www.youtube.com/embed/lB-F8WOHGeg" frameborder="0" allowfullscreen></iframe>

.. raw:: html

   <div style="position: relative; padding-bottom: 5%; height: 0; overflow: hidden; max-width: 100%; height: auto;">
     <iframe width="700" height="394" src="https://www.youtube.com/embed/bX9yWYTipO8" frameborder="0" allowfullscreen></iframe>

> - Once you've gone through all the QC, run the ``manual_correction.py`` script, as in an example below:

Reviewer: ```suggestion
- Once you finished the QC, run the ``manual_correction.py`` script, as in the example below:
```

.. code-block:: bash

>     manual_correction.py -config files.yaml -path-in ~/spine-generic/results/data -path-out ~/data-spine-generic

Reviewer: ```suggestion
    sg_manual_correction -config files.yaml -path-in ~/spine-generic/results/data -path-out ~/data-spine-generic
```

> - ``manual_correction.py`` script saves manually corrected files under derivatives/ folder (BIDS standard).

Reviewer: ```suggestion
- ``sg_manual_correction`` script saves manually-corrected files under ``derivatives/labels/`` folder, according to the BIDS convention.
```

Generate figures
----------------

> Generate figures based on the output csv files using ``generate_figures.py`` script. Run this script in ``/results``

Reviewer: Please update with new script name
